### PR TITLE
static: remove perl dependency

### PR DESCRIPTION
**Pull request by kubiko**
removing perl dependency from base snap

Signed-off-by: Ondrej Kubik <ondrej.kubik@canonical.com>

### CODE REVIEW

**Review by valentindavid — APPROVED** [review]
Looks good to me.

To be noted however that keywords in `sshd_config` are case insensitive. So it should also report lines with "hOstKeY". The bug was already here. But if you manage to improve the matching before merging that would be nice.

I also wonder if the `host_keys_required` could have been simplified if we used `sshd -T` instead of `sshd_config` since it reports default values.

File: static/usr/bin/core-sshd-host-keygen [hunk 1]
```diff
@@ -9,10 +9,8 @@ get_config_option() {
 	[ -f /etc/ssh/sshd_config ] || return
 
 	# TODO: actually only one '=' allowed after option
-	perl -lne '
-		s/[[:space:]]+/ /g; s/[[:space:]]+$//;
-		print if s/^[[:space:]]*'"$option"'[[:space:]=]+//i' \
-	   /etc/ssh/sshd_config
+	awk -v option=$option '$1 == option {print $2}' \
+        /etc/ssh/sshd_config
```
Comment (alfonsosanchezbeato) [inline static/usr/bin/core-sshd-host-keygen:13]:
```suggestion
	awk -v option=$option '$1 == option {print $2}' /etc/ssh/sshd_config
```

**Comment by xnox** [comment]
My understanding was that snapd was at some point calling some tool, from whithin the base snap, that was available as perl only. Has that been fixed in snapd to move to the other tool which is written in C?

(this is just a check to ensure we have the other largish perl dep dropped to be able to remove perl from the base snap)

**Comment by kubiko** [comment]
@valentindavid I have addressed now case insensitive  sshd config now as well

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks!

**Review by valentindavid — APPROVED** [review]

**Pull request merged**